### DISCUSSION TITLE
How to create a dynamic schema.

### DISCUSSION THREAD

**Discussion by laibulle** [ExpediaGroup/graphql-kotlin#1088] [Q&A]
Hello, I need to write a multi tenant GraphQL CMS like Contentful. I would append some query and mutation to a schema at runtime according to the client info.

How to mutate the schema at runtime for a given request ?

Best regards

**Comment by smyrick** [comment]
@laibulle If you are using `graphql-kotlin-spring-server`, I think it would be possible. You would have to reload the configuration beans that generate the schema though: https://github.com/ExpediaGroup/graphql-kotlin/blob/01db43b3f59862973d3c2ea7c7489b038390e08e/servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.kt#L63-L72

That would probably involve a bit more code outside of the libraries we provide so it is not something that is enabled "out of the box". Also this goes against some of the benefits of using a schema in the first place as something you can statically analyze in your builds or CI/CD pipeline.

If you are looking to have a merged GraphQL schema powered by another that is dynamic and that you don't control, I recommend you look at running a GraphQL gateway with [Apollo Federation](https://www.apollographql.com/docs/federation/)

Federation can be used to expose two separate GraphQL schemas as one schema, even if you don't use any of the type extension features.
